dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/helpers/HelperTypes.java

```diff
@@ -65,6 +66,21 @@ public static Expression createResultExpression(String id, Type type, Expression
         throw new ParsingException("Illegal result type in definition of '%s'", id);
     }
 
+    public static Expression createResultExpression(String id, Type type, Expression operand, IntUnaryOp op) {
+        if (isScalar(type)) {
+            return expressions.makeIntUnary(op, operand);
+        }
+        if (type instanceof ArrayType aType && aType.getElementType() instanceof IntegerType) {
+            List<Expression> elements = new ArrayList<>();
+            for (int i = 0; i < aType.getNumElements(); i++) {
+                Expression elementOp = operand instanceof ConstructExpr ? operand.getOperands().get(i) : expressions.makeExtract(operand, i);
```

Comment:
Is this case distinction necessary? I guess it gives you better-to-read code after parsing, but otherwise the `ExprSimplifier` would do the same optimization I believe.